### DISCUSSION TITLE
Soft delete functionality

### DISCUSSION THREAD

**Discussion by AzharUddinSheikh** [django-import-export/django-import-export#2022] [Ideas]
Wouldn't it be great if user want to soft delete the record !

as per [docs](https://django-import-export.readthedocs.io/en/latest/api_resources.html#import_export.resources.Resource.for_delete)

```
"""record get hard deleted if condition is true"""
def for_delete(self, row, instance):
    return row['delete'] == '1'
```

It would be much better if `for_delete` function have extra parameter for soft delete 

**Very Basic Solution**
```
"""record get soft deleted"""
def for_delete(self, row, instance, soft=False):
    is_deleted = row['delete'] == '1'
    if is_deleted and soft:
        instance.deleted = True
        return False # why False ? (if True unable to call `save_instance` method)
    return is_deleted
```

**Comment by matthewhegarty** [comment]
Thanks for raising this.  IMO soft delete functionality should be down to the user to implement in the way you have.  `for_delete()` is just a hook that is there to be overridden.  At this stage the library is mature (10 years+) and I would like to avoid adding extra functionality if we can avoid it, just because any new functionality adds overhead and potential for unintended bugs.  However if there are many other requests for this we can reconsider.
